iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/RemoveSubstation.java

```diff
@@ -0,0 +1,53 @@
+package com.powsybl.iidm.modification.topology;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.modification.AbstractNetworkModification;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.Substation;
+import com.powsybl.iidm.network.VoltageLevel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static com.powsybl.iidm.modification.util.ModificationReports.notFoundSubstationReport;
+import static com.powsybl.iidm.modification.util.ModificationReports.removedSubstationReport;
+
+/**
+ * @author Maissa Souissi <maissa.souissi
+ * at rte-france.com>
+ */
+public class RemoveSubstation extends AbstractNetworkModification {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RemoveSubstation.class);
+
+    private final String substationId;
+
+    public RemoveSubstation(String substationId) {
+        this.substationId = Objects.requireNonNull(substationId);
+    }
+
+    @Override
+    public void apply(Network network, boolean throwException, ComputationManager computationManager, Reporter reporter) {
+        Substation substation = network.getSubstation(substationId);
+        if (substation == null) {
+            LOGGER.error("Substation {} not found", substationId);
+            notFoundSubstationReport(reporter, substationId);
+            if (throwException) {
+                throw new PowsyblException("Substation not found: " + substationId);
+            }
+            return;
+        }
+        List<VoltageLevel> voltageLevels = new ArrayList<>(substation.getVoltageLevelStream().collect(Collectors.toList()));
```

Comment:
You can make it simpler : 
List<VoltageLevel> voltageLevels = substation.getVoltageLevelStream().collect(Collectors.toList());

Reply:
List<String> vlIds = substation.getVoltageLevelStream().map(VoltageLevel::getId).collect(Collectors.toList());
        vlIds.forEach(id -> new RemoveVoltageLevel(id).apply(network, true, reporter));

Reply:
done